### ISSUE TITLE
Improve small road highway tag guessing

### ISSUE THREAD

**Issue by atorger** [atorger/nvdb2osm#24]
For KLASS 8 and 9 these are usually better mapped to track in rural areas, and service in semi-urban areas. There's not enough information in NVDB tagging to now which is which though.

Maybe by looking at things like road density, distance to urban areas etc the script could make better guesses. Fully correct will not be possible, but the goal is to minimize the need for manual changes.

**Comment by NKAmapper** [comment]
I tried to experiment further with combinations of attributes. I have not found a specific attribute which denotes that a highway is used for forestry, but "Tillgänglighetsklass" is supposed to be registered on those highways, according to instructions. If you select highways according to the following criteria, I think you get 90% of what you want for track:

- Functional class 9 (class 8 are important highways in urban/mixed municipalities)
- "Tillgänglighetsklass" > 2 (class 1 and 2 are highway quality roads)
- No street name (excludes a few semi-urban highways)
- Not already tagged as Unclassified

The rest to be tagged as Service.

This seems to work across rural, urban and mixed municipalities. It is not perfect. For example you get "islands" of service at the end of tracks, but those are easily discovered during import.

**Comment by atorger** [comment]
Tillgänglighetsklass seems rather quirky in rural areas, points out very strange roads as 1 and 2 in Lycksele for example.

Anyway, I'm also working on this, which includes not only analyzing tags but geometry etc. It will be a few more days probably depending on time I get to spend on it, but I'll report back and see what I got. The goal is to get as many as possible correct over the whole spectrum rural->urban, but noting that 100% is not possible, so not trying to do something overly complicated as it will probably not perform better than a simpler model.

**Comment by atorger** [comment]
I have now added (a rather large) patch to "guess" upgrade track to unclassified and service, plus fix a quirk where short unclassified stumps are upgraded to residential.

This patch has most effect on rural areas. On outer areas of Göteborg and other semi-urban areas the change to look at if gatunamn is used (instead of övrigt vägnamn) and make those service instead of tracks took care of most roads.

On rural areas however the KLASS 8 level is highly unreliable, as well as tillgänglighetsklass, so the patch looks at properties of the geometry and makes upgrades from there. Roads that form connecting links become unclassified, then short stumps grouped together are identified as likely driveways, and if those are connected to a larger road it's upgraded to service. In a second pass other likely driveways nearby are identified and upgraded, and then in a final pass short tracks that links driveways to a larger road is upgraded to service. Log output from a run on Lycksele below. The result is not 100% perfect, but reduces the manual work significantly. The algorithm could be further refined, but I think this is a good start.

2021-03-19 22:08:45,739 - process - INFO - Upgrading short unclassified stumps to residential (if connected to residential)...
2021-03-19 22:08:45,782 - process - INFO - done (46 segments upgraded from unclassified to residential)
2021-03-19 22:08:45,783 - process - INFO - Making best guesses of upgrading highway=track to highway=unclasified or service...
2021-03-19 22:08:46,015 - process - INFO -   5134 tracks to be considered for upgrading (skipped 325 as they had gates, 896 as typical track names)
2021-03-19 22:08:46,113 - process - INFO -   69 tracks upgraded to unclassified as they form connecting links
2021-03-19 22:08:47,078 - process - INFO -   391 tracks upgraded to service as likely driveways connected to larger roads
2021-03-19 22:08:47,156 - process - INFO -   69 tracks upgraded to service as likely driveways near others
2021-03-19 22:08:47,341 - process - INFO -   42 tracks upgraded to service they form short links between roads and driveways
2021-03-19 22:08:47,342 - process - INFO - done (1312 of 5134 tracks upgraded to service)

**Comment by matthiasfeist** [comment]
cool. is that already on the main branch?

**Comment by atorger** [comment]
Yes, during this development phase everything goes in on the main branch :-). It's probably ok to run a new batch run now, but there's been quite many changes and not a whole lot of testing so we may see failures...

**Comment by matthiasfeist** [comment]
running the batch now

**Comment by NKAmapper** [comment]
Very interesting approach, and a better result. A bit refining is needed as there are still "islands" of unclassified and service.

A fair number of rural streets with homes are still getting highway=track. Would it make sense to first restrict highway=track to roads were the "Tillgänglighetsklass" attribute is present (disregarding the value, which is not consistent)? In that case only a few segments at the end of the road lacks any "Tillgänglighetsklass" and needs to be added to highway=track.

I looked mostly at Uppsala and Falun, which are mixed-use cases.

**Comment by atorger** [comment]
The problem with these type of algorithms is that there is just no information except the shape of the geometry, and if you tune it for one use case it will perform worse in another.

Note that I regard "track" wrongfully marked as "service" as worse than "service" wrongfully marked as "track", which leads to that I have excluded some aspects that will make service grasp over too much. I'll do a check on tillgänglighetsklass layer again.

**Comment by atorger** [comment]
Looked at Tillgänglighetsklass, it cannot be used as a exclusive cutoff, but it could be used in a weighting algorithm.

Tillgänglighetsklass is indeed missing on driveways, but also on shorter forestry roads and some longer. I think though that if looking if there are nearby "safe" driveways one can with quite good confidence upgrade shorter roads.

There are a few "impossible" cases, and that's single isolated dwellings with a single short (or longer) road, these are exactly the same as the many short forestry roads for roadside clearcuts. However, as soon as there are pairs closeby, one can quite safely assume it's driveways as it's very rare for forestry roads to make short stumps close together, as there's no need.

**Comment by atorger** [comment]
I ran a test on Åstorp which i small municipality in south of Sweden, quite good for making fast testing on mixed-use.

By converting tracks to service which lacks Tillgänglighetsklass but are less than 1000 meters long and have a service/residential road within 300 meters and going through all of the roads affected by this rule I got ~75% right, and ~25% wrong, and a few that should be unclassified or residential instead. Similar numbers in Lycksele, although I think it will be a higher number correct there. The problem is that the bad ones turn up in places you are less likely to look, ie away from urban areas in lesser mapped places. When working with places like Göteborg, Uppsala or Falun basically every road is already mapped in some shape or form. Often with not so good geometry and lacking in tags, but it's there and need manual merging.

That's why I think it's actually more important to get things right in less dense areas, than in more dense ones, and I think it's better to keep too many tracks than upgrade too many to service.

Not sure how much value it is spending a lot more time on this though, it takes a lot of time tuning and testing these algorithms and it's a delicate balance, I'll probably put this to the side for a while.

I have recently working with importing semi-urban areas, and it's a lot of work. I have started with a older version of the script so more highways are wrong type, but I can say it's not there most time is spent anyway. One thing is that there's a lot more driveways mapped in OSM than exist in NVDB, and all these need re-attaching. And there's a lot of alignment errors of buildings and farmlands and other natural areas that needs correcting to fit roads, and one cannot just bulk-move as the alignment error varies over the surface.

**Comment by matthiasfeist** [comment]
the batch completed and at least it looks like all osm files were generated correctly: https://nvdb-osm-map-data.s3.eu-north-1.amazonaws.com/index.html

**Comment by matthiasfeist** [comment]
I just checked for myself the area where I live and I think there is a problem that I just found:
https://www.openstreetmap.org/#map=19/59.36117/17.96452 the road "landsvägen" is supposed to be a "lokalgata stor" which I think should translate to a tertiary road? it's currently marked as "residential" and that's definitely not true :)

**Comment by atorger** [comment]
It's a tradeoff, making lokalgata stor to tertiary in big cities makes them really overflow with tertiary. While that might be considered okay if we would start from scrach, it's not how they are mapped currently. Below is the current translation rules for city roads (those that exist in the gatutyp layer).

As you see only "Huvudgata" is translated to tertiary, and only if Funktionell vägklass is 5. For other roads also vägklass 6 becomes tertiary.

I've tuned these translations a bit differently, but anyway you do it there will be quite much mismatch from how it's mapped in OSM currently. When there is a difference, I change to what's in OSM while merging, unless it's obviously wrong. When discussing this a while back on the Facebook group it was very clear that the traditionalists is not keen at all of the idea of using fixed translations from NVDB classes to primary/secondary/tertiary etc, but they want this to be a mapper judgement, so I don't think we can make a translation that all will accept. So I think we will have to live with having to adjust these types quite often.

            if gatutyp == "Övergripande länk":
                raise RuntimeError() # should already be handled
            if gatutyp == "Huvudgata":
                if klass <= 1:
                    tags["highway"] = "trunk" # 0, 1
                elif klass <= 2:
                    tags["highway"] = "primary" # 2
                elif klass <= 4:
                    tags["highway"] = "secondary" # 3, 4
                elif klass <= 5:
                    tags["highway"] = "tertiary" # 5
                else:
                    tags["highway"] = "residential"
            elif gatutyp == "Lokalgata stor":
                tags["highway"] = "residential"
            elif gatutyp == "Lokalgata liten":
                tags["highway"] = "residential"
            elif gatutyp == "Kvartersväg":
                tags["highway"] = "service"
            elif gatutyp == "Parkeringsområdesväg":
                tags["highway"] = "service"
                tags["service"] = "parking_aisle"
            elif gatutyp == "Infartsväg/Utfartsväg":
                tags["highway"] = "service"
            elif gatutyp == "Leveransväg":
                tags["highway"] = "unclassified"
            elif gatutyp == "Småväg":
                tags["highway"] = "unclassified"
            else:
                raise RuntimeError("Unknown gatutyp %s" % gatutyp)

**Comment by NKAmapper** [comment]
I understand the frustration :)  I made a few [examples](https://www.jottacloud.com/s/0592631d082283b49939bb929c3050af021) with Tilgänglighetsklass. It seems to be a sufficient universe from which to get track. Of course the service "islands" at the end of branches would have to be included. And I am sure your more sophisticated method could identify a few unclassified and service roads among the tracks.

I just think there is too much guessing going on now. It would be clearer to base tagging on what is actually contained in NVDB (for good and for worse, just like primary/secondary etc), and I have found no other attribute than Tilgänglighetsklass to denote forestry roads. Sure it will not be perfect. That is why we do a manual import.

My experience is that importing is equally useful in semi-urban areas because the existing mapping is often with a big offset and with missing maxspeed, street name etc. Downtown areas in cities, on the other hand, are often well mapped because more mappers live there.

You could of course also have a special tagging variant for import needs in the north of Sweden.

**Comment by atorger** [comment]
I agree that it's useful to map already mapped areas, but what you get there is preset highway types.

No-one should ever *delete* existing roads and replace them with imported. It's just plain wrong if we suggest that it would be a valid workflow, and if anyone is caught doing that they should not be allowed to do imports. The old should be *modified* to new, either manually or using replace geometry tool. So why is modify instead of delete relevant in this context? When you modify you get clear feedback to set your highway type to either keep the old or change to the script's best guess.

This is not just about service vs track. It's about service, track, residential and unclassified, in semi-urban places all these four types may have same type of NVDB tags. You can't just look at the script output and decide if you like that or not, you must also look at what is pre-existing in OSM, and make a fair judgement of what's the best call, not just pick script result because that's the easiest. When doing that, it's clear that semi-urban places require careful work.

That said I think that there's quite of lot of bad highway choices in OSM already, so I correct it a lot too.

One can try that discussion rely on a static translations with the traditionalists :-). They couldn't care less about what is in NVDB. If NVDB provides wrong information, they are not happy to accept that just because it's in NVDB. They just look at what we want to be in OSM, not looking at NVDB *at all*. And I agree with them. We should not change our ways to map just to match what can be done with NVDB. We should look at how these ways should be tagged in OSM and Swedish tradition, and only then see if NVDB can match that. It can't.

The purpose of a guessing algorithm is to minimize the need to manually change. If the guessing algorithm causes a larger need to change than using statically incorrect conversions from various unreliable NVDB tags, then I agree it's probably better using a static conversion.

I think a problem here is though that we don't agree about the degree of correctness of a static conversion, I think you believe it's more correct than I do.

Let's look at an example from Uppsala.

This is currently in OSM today:
![image](https://user-images.githubusercontent.com/13962807/111904857-34528100-8a49-11eb-82ca-e882ffa22612.png)

This is static conversion output from your script (don't know if I have the latest):
![image](https://user-images.githubusercontent.com/13962807/111904886-50562280-8a49-11eb-8d15-1a990e547939.png)

This is the guesswork conversion from my script from Matthias latest batch run:
![image](https://user-images.githubusercontent.com/13962807/111905663-0c651c80-8a4d-11eb-9877-5eadff840ca8.png)


I want to make changes to all these, also what's pre-existing in OSM.

Let's look at Örkälla, southwest of Skyttorp. It's residential in OSM, service in your script and track in mine. Residential is best call, but there's no tagging in NVDB for that, that's why our scripts get stuck on track/service.

Look at the forestry roads to the east. Here I say that my script has even better output than what's currently in OSM, which has too many unclassified. You don't want have gated roads as unclassified unless you have local knowledge and knows that the gate is permanently open or there's some other special reason. When it comes to quarries these are often abandoned, and type of road depends on if it's in active use or not.

Again, track is not tractor track exclusively in Swedish mapping tradition, it's also less important unpaved roads, which we have a lot of.

On secondary vs tertiary my script matches OSM, and the reason for that is that I use FPV-klass rather than funktionell vägklass for numbered roads which better match what OSM has now (but still not always...)

On the service roads, my script has made few mis-upgrades, but obviously misses quite many, and the reason of that is because it becomes too hard to differ between what's most likely roads to houses and what's forestry/argiculture tracks. However, these are generally in clusters which fit in your edit view, and I think it's better to have issues there than on long-winding roads.

For this example I certainly prefer working with the output the guesswork output makes rather than static rules.

Below a zoomed in example from Gödåker, which is seen center left on the others, after I have manually pre-processed the highway types:

![image](https://user-images.githubusercontent.com/13962807/111905888-4125a380-8a4e-11eb-98ae-cf409fdc15f8.png)

Some short ones needed upgrading to service, but there's also some split work, Fredrikslund starts as service and transitions into track, ending with a turning circle (which needs adding, no turning circles on small roads in NVDB)

Gödåker and Lilla Gödåker, service, unclassified or residential? I have made it unclassified as there are several houses reached and the road is not specifically to one house. One might argue this should be residential, and I'm not entirely sure on this, but my current strategy which seems to fit mapping tradition well is to use unclassified unless there's a certain level of density when it becomes residential. Likewise Husby is split between unclassified and service.

**Comment by matthiasfeist** [comment]
Haha. Amazing thanks for the input. I just picked something close to me at random and sure enough found the hardest problem ;) 
I really enjoy all the in-depth explanations. thanks a lot!

**Comment by NKAmapper** [comment]
It seems you have not looked at my examples (which by the way only was produced to show the extent of tracks, not a complete end product). And I also seem to have failed to explain my approach. But I give up now.

**Comment by atorger** [comment]
Oops, I did miss the examples link, sorry for that, looking into it!

**Comment by atorger** [comment]
Here's a screenshot from the Tillgänglighetsklass script run, again sorry for missing that:

![image](https://user-images.githubusercontent.com/13962807/111909926-7dadcb00-8a5f-11eb-9fc5-69e0ecac7710.png)

**Comment by atorger** [comment]
On a related subject: what I think we will have to do if we want to formalize this in an import process "available to all on full scale" is to start in a different end: making a document describing how highway types should be mapped in Swedish conditions, and specifically focusing on residential/unclassified/service/track, and get that accepted by the Swedish OSM community.

This document should not mention NVDB at all. It's just OSM on OSM's own conditions. In theory the OSM wiki should already have all the answers, but Swedish conditions are a bit special and there's an existing mapping tradition to take into account etc, and there are lots of corner cases.

I've learnt that examples is best, the document would contain similar to the screenshots above with different type of conditions and which types to use when.

When we have that established, the next step is to see how well this maps to what NVDB can provide. With that document in place, we have a common reference and it's easier to argue for what's correct in an OSM context.

Then one can start working on tuning the script to make as good job as possible to match that document guidelines, and then make a import process which describes caveats where one need manual changes etc.

I think it's the wrong way to look at what NVDB can do, and then convince ourselves that it is the best way to use OSM highway types. Even if NVDB functional class data would be 100% correct and consistent, which it isn't especially in the 7-9 range, highway types is not really defined in the same way as NVDB uses functional classes.

Sure for the range trunk to tertiary, one could argue for a static conversion, because there functional class maps to the highway types pretty 1 to 1, it's just that OSM mappers have in many places made a bit different decisions than NVDB, and NVDB itself has slightly different functional classes in different layers on these roads (depending on traffic type etc), so even there I think that manual adjustments are to be expected.

However, when it comes to residential, unclassified, service and track it becomes more complicated, it's sort of both functional class and use based. NVDB has no information if a road is used as a driveway. It indeed has information about residential, but much more area-based to maxspeed=50 areas and it misses quite a lot of villages that traditionally get residential in OSM. Actually I think I may be under-using residential myself currently, it's one thing I'll take up with the community. Service is about accessing something. And then we have track which is used a bit special in Swedish conditions, as we have this huge forestry network.

Making such a document will be lot of work though, and many discussions of people of differing opinions. I'm not totally against the idea of making such a document myself at some point, but it's probably not going to happen this year.

For my own purpose, the script is mature enough to entertain me with import work for years. As long as I am the single user that uses it on scale, I have lots of eyes on me, and informal explanations and discussions of what I'm doing is enough.

**Comment by atorger** [comment]
Here's a test with both tillgänglighetsklass *and* guesswork track upgrade active:
[www.torger.org/anders/downloads/uppsala-new.osm](url)

Note that I haven't run it with all layers so there's info missing, but most related to highway tag is there.

**Comment by atorger** [comment]
So I have added --small_road_resolve where you can specify prefer_service, prefer_track, or its static variants. Without static, it will on top run the guesswork upgrading of remaining tracks to unclassified/service (ie after static rules have been applied)

This is "prefer_service_static", should be most similar to NKAs, download here: [prefer_service_static](https://www.torger.se/anders/downloads/uppsala-prefer_service_static.osm)
![image](https://user-images.githubusercontent.com/13962807/111914135-11d45e00-8a71-11eb-91a7-8f6bc6979a40.png)

This is "prefer_service", ie some track upgrades applied, not big difference in this view, but one can see some tracks have become unclassified,
[prefer_service](https://www.torger.se/anders/downloads/uppsala-prefer_service.osm)
![image](https://user-images.githubusercontent.com/13962807/111914354-e0a85d80-8a71-11eb-87b6-fff30746df40.png)

And finally this is "prefer_track", ie same as the latest batch run
[prefer_track](https://www.torger.se/anders/downloads/uppsala-prefer_track.osm)
![image](https://user-images.githubusercontent.com/13962807/111914366-e9009880-8a71-11eb-9d28-16d676d5bcbc.png)

**Comment by atorger** [comment]
Just let me know which one you prefer as default setting, or if you want something else.

In the end I'm 100% sure that we will *not* get acceptance to use NVDB-derived highway tags, guessed or static, straight off as-is. Manual inspection and adjustment will be an absolute requirement on the importer. Just as using modify instead of delete followed by insert is an absolute requirement.

And then the task for the script is just about making an output that the importer finds effective to work with, and there we will see different preferences I suppose.

My preference is as follows: for roads that need tag adjustment I prefer when they are clustered (for example too many tracks inside a village is easier to correct than too many service spread out in the forestry network), and if I do mistakes I prefer to leave one too many with track tag than one too many service or unclassified tag. The reason for that is I think it's worse indicating that a road leads to something (service) when it doesn't than the other way around. On track vs unclassified I think when mistakes occur it's better with under-classing than over-classing.

By having a few different rendering options maybe we can cover most preferences.

What should the monthly batch run use though? Maybe two of the most popular. One reason to have two is to make it *really* clear for those that don't read docs that highway tag isn't set in stone.

**Comment by matthiasfeist** [comment]
I wonder if it would be an option to leave the highway tag out completely for the most "disputed" tags? That way they have to be set manually? Maybe that's too much work when actually importing the file...

Another option would be to set the flag by "kommun". So e.g. we know that setting X works best in Stockholm but setting Y works best in rural areas.

By the way, I have tinkered a bit further and now created a pipeline that produces vector tiles.
Have a look here: https://nvdb-osm-map-data.s3.eu-north-1.amazonaws.com/map.html#7.8/64.489/18.735 Currently on this map you see the differences in the tag `highway`. Additionally, when you zoom in a bit further, you get an overlay of the geometry that the nvdb2osm script outputs with the highway tag on the line. Maybe that helps.
My plan is to have this map toggleable so you can toggle if you want to see where the most differences are in lanes, maxspeed, highway etc. 
But maybe I'll leave the highway tag out completely if it does maybe more harm than good.

**Comment by atorger** [comment]
I've also thought about leaving out the highway tag from script output. When working with previously mapped areas it's in theory quite practical actually, as you then never need to resolve highway tag conflict, with replace geometry you will automatically use what was previously there. However, it becomes a little bit messier to work with visually in the editor as the geometry then just becomes gray lines. It's also extremely easy for the user to remove highway tag from the layer if he or she wants, just select and delete.

If we are afraid that people are going to use the pre-made batch run data carelessly one could indeed leave it out of those files just for that reason. *However*, I'm not too worried about that, I think it's more about how we lay out the documentation, most want to do a good job. If one is worried about careless highway tag override I would be much more worried about people just deleting everything in view and pasting in the new layer, or messing up route relations, or turn:lanes on the bigger roads etc. There's so many wrongs you can do in addition to highway tag which we can't do anything about, except policing.

In terms of visualization of your comparison map (really nice by the way, just tested the new version), I'm thinking that maybe just differing between cycleway/footway/path and then all other highways is enough, and perhaps not with text which tend to distract overviews, but instead with different line width or dashed vs solid or something. When scanning for differences to find something suitable place to fix one will not look for mismatching highway tags, but things like maxspeed and geometry offsets.

**Comment by matthiasfeist** [comment]
Yeah you're right, maybe we're overly worrying...
Good idea. I'll put more effort down into the strengths of NVDB which is geometry and speeds, lanes etc and build the diff map around that.

I guess it would be really nice if you and @NKAmapper could come to a conclusion around the tagging so that we then can prepare the proposal for the import (for which NKA has already made a nice wiki page)... 
My primary intention is really just to help as many people to improve OSM rather quickly :)

**Comment by atorger** [comment]
With risk of sounding a bit negative again, I think it will be tough to get a lot of people to actually help do the import work at least on short notice, so I would not feel too much in a hurry of the import proposal. This script has been out for two months, I've talked about it and the way I have put it to use frequently on the OSM Sweden facebook group, and still it's only the three of us that's active here on github. I was a bit naive and thought that there would be a bit more interest. There's no lack of appreciation of the import work I've already done, it's not that, but it's quite clear that among the active mappers there's no hunger for using import scripts to make their job quicker.

Part of the reason is that many mappers feel that it's a bit too difficult and they are afraid of making mistakes. I'd say that it's actually not that difficult, but if you never have used JOSM before there's a bit of a threshold for sure.

I've thought about making a "youtube training video" with JOSM and the import script and merging techniques. Text documentation gets a bit dense, the readme file to this script is already pretty crazy. I think a video could be very good way to get new mappers take the step and join the work. I've done some lame but effective instructional videos before for other reasons, so it's something I could do, but probably not until after summer if so.

There are a handful of advanced mappers in the active Swedish community that have the know-how already of course. I know of a few that already use NVDB since quite a while, I got my initial pointers to making my script from them, but they have their own scripts and methods, and still after knowing about my considerably more advanced and detailed script (ie much richer tagging) they still use their own methods and haven't even tried this script or commented. The thing is that I think fewer than we think has a specific desire to make OSM a complete working end product for whole of Sweden, but rather has pretty local interests or just find like mapping as a relaxing activity where the speed of mapping or richness of tags is not really that interesting.

So even with a fancy import guide, if we don't do anything more than that to attract mappers I think there's a big risk that there will only me and NKA doing all the full-coverage importing... to me that wouldn't be a failure though, I'm happy that I now can map much faster than I did before, I enjoy the mapping activity more this way, but sure I can't realistically finish Sweden on my own. In time I can cover my local interests though.

**Comment by atorger** [comment]
FYI: I collect edit guidelines based on what I use myself and on discussions in the facebook group in the README file. This should probably be refined with image examples and moved to an appropriate guideline document on the OSM wiki and be peer reviewed at some point. I've just kept it there for now for my own convenience. Anyhow, under the anonymous heading "Övrigt att tänka på" there's guidelines of how to relate to the script's highway tags and other things.

**Comment by matthiasfeist** [comment]
@atorger something completely different: I just found this: https://nvdb-osm-map-data.s3.eu-north-1.amazonaws.com/map.html#12.08/59.96192/15.85696 Is your script running amok? 😆 seems like it generates quite some chaos...

**Comment by matthiasfeist** [comment]
On the topic of your longer message: I do understand that it will probably not be a huge group that does the import and that's fine. Maybe a youtube tutorial is actually a really good idea. Part of me programming the pipeline and the map is the hope that this would lower the barrier of entry for some people. (OK and because it's actually really fun to tinker with these things 😉)

But as I understand the OSM customs, we should have an import plan ready and OK-ed by the community, right? 
I would like to help with this as much as possible but this is a completely new field to me. I'd like to have @NKAmapper's guide for this as much as possible as you have already successfully driven such projects.

**Comment by atorger** [comment]
@matthiasfeist that's Fagersta kommun which looks ok in JOSM if you download it from the batch run page. If something like that would happen in the script it would never complete, so there's something going on somewhere else in the pipeline.

Yes I think your map will not only lower the barrier, but also make more people interested.

Yes we should have an import plan, especially if more people starts getting involved. I realize that it may be a bit problematic or at least uncomfortable to publish your "lower barrier of entry" project without having a import plan in place. I guess I could help with some text, just point me where I should write :-).

I don't know how you roll these things, but I guess one can invite people from the international mapping community to help, afterall many of the most active mappers in Sweden are not Swedes, top list here: https://osmstats.neis-one.org/?item=countries&country=Sweden. If we do that I think it becomes even more important with a solid import plan.

**Comment by matthiasfeist** [comment]
:) yes it seems to be a bug in my script. Damn...

**Comment by atorger** [comment]
Closing this. If we need further refinements, new algorithms etc, open a new issue.
